src/file.rs

```diff
@@ -261,32 +292,44 @@ mod tests {
     }
 
     #[test]
-    fn as_posix() {
-        result_eq(p!(".").join("..").as_posix(), Ok("./..".to_owned()));
-        result_eq(p!(".").join("/").as_posix(), Ok("/".to_owned()));
-        result_eq(p!("foo").join("bar").as_posix(), Ok("foo/bar".to_owned()));
-        result_eq(p!("/foo").join("bar").as_posix(), Ok("/foo/bar".to_owned()));
+    fn as_posix_relative() {
+        result_eq(
```

Comment:
nit not related to these changes

`result_eq` should have a `#[track_caller]` attr